src/backend/marsha/core/api/live_session.py

```diff
@@ -317,61 +428,36 @@ def list_attendances(self, request, video_id=None):
         The same one will then be used for all the students on this list.
         Serializer is cached so the listing don't get recalculated too often
         """
-        prefix_key = f"{VIDEO_ATTENDANCE_KEY_CACHE}{self.request.resource.id}"
+        video_id = self._get_related_video_id()
+
+        prefix_key = f"{VIDEO_ATTENDANCE_KEY_CACHE}{video_id}"
         cache_key = (
             f"{prefix_key}offset:{self.request.query_params.get('offset')}"
             f"limit:{self.request.query_params.get('limit')}"
         )
-        if cached_data := cache.get(cache_key, None):
-            return Response(
-                OrderedDict(
-                    [
-                        ("count", cached_data.get("count")),
-                        ("next", cached_data.get("next")),
-                        ("previous", cached_data.get("previous")),
-                        ("results", cached_data.get("results")),
-                    ]
-                )
-            )
+        if (cached_data := cache.get(cache_key, None)) is not None:
+            return Response(cached_data)
 
         try:
-            video = Video.objects.get(pk=self.request.resource.id)
+            video = Video.objects.get(pk=video_id)
         except Video.DoesNotExist as exception:
             raise Http404("No resource matches the given query.") from exception
```

Comment:
Can we use here `get_object_or_404` ? I don't remember the reason we didn't use it.